react/features/youtube-player/components/native/YoutubeLargeVideo.js

```diff
@@ -116,84 +107,226 @@ type Props = {
     youtubeId: string
 };
 
-const YoutubeLargeVideo = (props: Props) => {
-    const playerRef = useRef(null);
+/**
+ *
+ * Implements a React {@code Component} for showing a youtube video.
+ *
+ * @extends Component
+ */
+class YoutubeLargeVideo extends Component<Props, *> {
+    /**
+     * Saves a handle to the timer for seek time updates,
+     * so that it can be cancelled when the component unmounts.
+     */
+    intervalId: ?IntervalID;
 
-    useEffect(() => {
-        playerRef.current && playerRef.current.getCurrentTime().then(time => {
-            const { _seek } = props;
+    /**
+     * A React ref to the HTML element containing the {@code YoutubePlayer} instance.
+     */
+    playerRef: Object;
+
+    /**
+     * Initializes a new {@code YoutubeLargeVideo} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+        this.playerRef = createRef();
+
+        this._onReady = this._onReady.bind(this);
+        this._onChangeState = this._onChangeState.bind(this);
+    }
 
+    /**
+     * Seeks to the new time if the difference between the new one and the current is larger than 5 seconds.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate() {
+        const playerRef = this.playerRef.current;
+        const { _seek } = this.props;
+
+        playerRef && playerRef.getCurrentTime().then(time => {
             if (shouldSeekToPosition(_seek, time)) {
-                playerRef.current && playerRef.current.seekTo(_seek);
+                playerRef && playerRef.seekTo(_seek);
             }
         });
-    }, [ props._seek ]);
+    }
+
+    /**
+     * Sets the interval for saving the seek time to redux every 5 seconds.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidMount() {
+        this.intervalId = setInterval(() => {
+            this.saveRefTime();
+        }, 5000);
+    }
+
+    /**
+     * Clears the interval.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        clearInterval(this.intervalId);
+        this.intervalId = null;
+    }
 
-    useEffect(() => {
-        props._onWideScreenChanged(props._isWideScreen);
-    }, [ props._isWideScreen ]);
+    /**
+     * Renders the YoutubeLargeVideo element.
+     *
+     * @override
+     * @returns {ReactElement}
+     */
+    render() {
+        this.setWideScreenMode(this.props._isWideScreen);
```

Comment:
doing this on the render is not a good idea, can we do it based on an event instead?